packages/delegation-process/src/services/delegationService.ts

```diff
@@ -148,6 +150,16 @@ export function delegationServiceBuilder(
       assertDelegatorIsProducer(delegatorId, eservice);
     }
 
+    if (kind === delegationKind.delegatedConsumer) {
```

Comment:
```suggestion
    } else {
```
Since these checks are mutually esclusive I would use an `else`

Reply:
Even better, I propose to use an exhaustive TS pattern match, so that we make sure all delegation kinds are handled, even in case we add new ones